state/etcd/etcd.go

```diff
@@ -80,6 +81,15 @@ func (e *Etcd) Init(_ context.Context, metadata state.Metadata) error {
 
 	e.keyPrefixPath = etcdConfig.KeyPrefixPath
 
+	switch etcdConfig.Version {
```

Comment:
I think the version should not come from the metadata. Instead, we should have a separate version of `NewEtcdStateStore` that returns a *Etcd configured with the version we want. Then, in the runtime we register the 2 components separately, and just use the different `New*` method

Reply:
What's the reason for doing that?
Having it in metadata means runtime always sends the version, rather than only sending this metadata on this specific state store specific component.

Reply:
@ItalyPaleAle I have now updated the PR so that the Version is passed in as part of the `metadata.Base` struct as a dedicated field. This seems a more correct place to be passing information.

After merging, dapr will need to be updated so that it always bases the `Version` string, along with the `Name` of the component.

Reply:
I was hoping for an actual separate `New` method mostly to keep hygiene and force us to do things right.

1. It seems more appropriate to do the “version routing” decision in the runtime and not in the component. The `registry` code already supports [registering versioned packages](https://github.com/dapr/dapr/blob/master/pkg/components/state/registry.go#L42) (using `name/v2` format, although we haven’t tested that thoroughly)
1. In the future there’ll be a time where we will have a “v2” of a component that is an actual full rewrite, so it will be in a separate package. This seems like a good opportunity to do the work to make sure we can support that.

Reply:
I've updated the PR to now use two different `New` funcs to select the schema version. PTAL @ItalyPaleAle

Reply:
Thanks!